File: Backend/Geo_Cortex/Geo_Cortex_Assistant/app/routers/jobs.py
from __future__ import annotations

from typing import Any, Dict, Optional

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.database import SessionLocal
from app.models.dbmodels import Job
from app.services.governance import audit_log, feature_enabled


router = APIRouter(prefix="/jobs", tags=["jobs"])


def get_db():
    try:
        db = SessionLocal()
        yield db
    finally:
        db.close()


@router.get("/{job_id}")
async def get_job(job_id: str, db: Session = Depends(get_db)) -> Dict[str, Any]:
    if not feature_enabled("jobs"):
        raise HTTPException(status_code=403, detail="Jobs API is disabled by data governance policy.")
    j: Optional[Job] = db.query(Job).filter(Job.id == job_id).first()
    if not j:
        raise HTTPException(status_code=404, detail="Job not found")
    audit_log("jobs_get", {"job_id": job_id, "status": j.status})
    return {
        "id": j.id,
        "type": j.type,
        "status": j.status,
        "created_at": j.created_at.isoformat() if j.created_at else None,
        "updated_at": j.updated_at.isoformat() if j.updated_at else None,
        "progress": j.progress,
        "message": j.message,
        "result": j.result,
        "error": j.error,
    }

File: frontend/src/api/jobs.ts
import { httpJson } from "./http";

export type JobResponse = {
  id: string;
  type: string;
  status: "pending" | "queued" | "running" | "succeeded" | "failed" | string;
  created_at?: string | null;
  updated_at?: string | null;
  progress?: number | null;
  message?: string | null;
  result?: any;
  error?: string | null;
};

export async function fetchJob(baseUrl: string, jobId: string): Promise<JobResponse> {
  const b = baseUrl.replace(/\/$/, "");
  return await httpJson<JobResponse>(`${b}/jobs/${encodeURIComponent(jobId)}`, { method: "GET" });
}

